### PR TITLE
chore(msa): update on-initialize benchmarks

### PR DESCRIPTION
**Pull request by enddynayn**
# Goal
Update on-initialize benchmarks to use
T::MaxSignaturesPerBucket for calculating
benchmarks. This update allows benchmarks
to be calculated based on production configuration.

Update how signatures are calculated to decrease
the time that it takes to run benchmarks.

References #668

### CODE REVIEW

**Review by shannonwells — APPROVED** [review]
Awesome

File: pallets/msa/src/benchmarking.rs [hunk 1]
```diff
@@ -205,11 +198,13 @@ benchmarks! {
 	on_initialize {
 		// we should not need to max out storage for this benchmark, see:
 		// https://substrate.stackexchange.com/a/4430/2060
-		let m in 1 .. 3_000;
+		let m in 1 .. T::MaxSignaturesPerBucket::get();
 		for j in 0 .. m {
 			let mortality_block = 49 as u32;
```
Comment (aramikm) [inline pallets/msa/src/benchmarking.rs:203]: what is the reason behind keeping `mortality_block` fixed?

**Review by aramikm — APPROVED** [review]
Looks good!

File: pallets/msa/src/benchmarking.rs (hunk 1 above)
Comment (enddynayn) [reply pallets/msa/src/benchmarking.rs:203]: Keeping it constant makes it easy since making mortality_block more dynamic would not impact the benchmarks.

**Pull request merged**